### PR TITLE
Plotter service refactored with pydantic

### PR DESCRIPTION
**Pull request by MrToino**
Plotter service refactored, now supporting pydantic BaseModel. Optimized plotter workflow.

### CODE REVIEW

File: application/src/components/WorkspacePlot.vue [hunk 1]
```diff
@@ -36,8 +36,20 @@ export default defineComponent({
                 return;
             }
 
-            formData.append('file', this.file);
-            
+            formData.append('rawData', this.file);
+        
+            const options = JSON.stringify({
```
Comment (luisferreira32) [inline application/src/components/WorkspacePlot.vue:41]: Should we keep this hardcoded here for now?

File: services/plotter/server.py [hunk 2]
```diff
@@ -1,49 +1,70 @@
...
+from services.utils.exceptions import InvalidRequestError
 
 # To ensure we can plot in different threads
 # https://matplotlib.org/stable/users/faq/howto_faq.html#work-with-threads
 # https://matplotlib.org/stable/users/explain/backends.html#selecting-a-backend
 matplotlib.use("agg")
 
-with open(os.path.join(".", "schema.json")) as f:
-    PLOTTER_OPTIONS_SCHEMA = load(f)
+# with open(os.path.join(".", "plotter\\schema.json")) as f:
```
Comment (luisferreira32) [inline services/plotter/server.py:19]: Remove this two lines, no need to leave comments like this

File: services/plotter/server.py [hunk 3]
```diff
@@ -1,49 +1,70 @@
...
-        data, options = parse_request(rawData, rawOptions, PLOTTER_OPTIONS_SCHEMA)
-        plot_img = service(data, options)
-    except (InvalidRequestError, ValidationError, ValueError) as e:
+        data = read_csv(rawData.file, sep=",", index_col=0)
+        plot_img = service(data, rawOptions)
+    except (InvalidRequestError, ValueError) as e:
         raise HTTPException(status_code=400, detail="Bad request: " + str(e))
     except Exception as e:
-        raise HTTPException(status_code=500, detail="Oops: " + str(e))
+        raise HTTPException(status_code=500, detail="Oops my bad: " + str(e))
```
Comment (luisferreira32) [inline services/plotter/server.py:56]: my mad? 👀

**Review by luisferreira32 — APPROVED** [review]
lgtm, just a couple comments

File: application/src/components/WorkspacePlot.vue (hunk 1 above)
Comment (MrToino) [reply application/src/components/WorkspacePlot.vue:41]: For now we will let this hardcoded just for testing purposes. Later on we will deal with it.

File: services/plotter/server.py (hunk 2 above)
Comment (MrToino) [reply services/plotter/server.py:19]: Thanks for the reminder! All done

**Pull request merged**